Make https evcc hosts work

RustyDust opened the pull request:
While working on a quick hack (fix) for the new `chargerPhasesSwitching` entity I found that ha-evcc wouldn't update my stats. Digging deeper I found lots of `"ADD-API: write_values failed ..."` error messages.

That's because internally ha-evcc only uses the actual host definition without the scheme ("http://" or "https://") and expects the evcc instance to be reachable via http://. Unfrotunately mine isn't so I reworked `config_flow.py` and `pyevcc/__init__.py` a bit to make them work with full scheme+host URLs.

I also left in all the changes suggested by my IDE (like direct access to private class members and such), feel free to cherry pick what you like and throw away the stuff that doesn't suit you.

marq24 commented:
Thanks for your PR - plenty of your commits tackle my limited python knowledge (at least I assume so)... I will probably will cherry pick most of them [for some I am currently struggling understanding the 'why' (e.g. renaming of `service` to `services`)]...

For the main feature (`http + port` vs. `https`) I might like to follow a different approach in the setup-flow (adding an additional checkbox) - but I have to make up my mind...

marq24 commented:
I have now cherry picked now most of your commits (directly from your repo)... and added some additional tweaks:
- added support for old configurations (all old evcc setups does not contain the schema in the host setting) - so this must be checked and added if not present
- when https is used, then IMHO the integration should be able to handle self-signed certificates (which requires an additional parameter in each request)

I left out a few of your commits - the only reason is my personal preferences concerning code readability and consistently with the other integrations I am maintaining.

RustyDust commented:
> I have now cherry picked now most of your commits (directly from your repo)... and added some additional tweaks:

Cool, thank you!

> * added support for old configurations (all old evcc setups does not contain the schema in the host setting) - so this must be checked and added if not present

Ooops. Should have thought about that. Sorry for the oversight :(

> * when https is used, then IMHO the integration should be able to handle self-signed certificates (which requires an additional parameter in each request)

I'm so used to only using "real" certificates, I tend to forget that there are self-signed ones out there. So again: good catch!

> I left out a few of your commits - the only reason is my personal preferences concerning code readability and consistently with the other integrations I am maintaining.

No problem there, it's your repo and your code. I'm happy that the https stuff is working ;)
Hope you don't mind me closing this PR then.
